src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    private BulldozerProducer bulldozerProducer = new BulldozerProducer();
+    private ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+    private TruckProducer truckProducer = new TruckProducer();
+    private List<? extends Machine> machineArrayList = new ArrayList<>();
+
+    @Override
+    public List<Machine> getAll(Class type) {
```

Comment:
```suggestion
    public List<Machine> getAll(Class<? extends Machine> type) {
```
it's a bad idea to use raw types